core/shared/src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -2,227 +2,244 @@ package zio.prelude.fx
 
 import zio.internal.Stack
 import zio.prelude._
-import zio.{ CanFail, NeedsEnv }
+import zio.test.Assertion
+import zio.{ CanFail, Chunk, NeedsEnv }
 
 import scala.annotation.switch
 import scala.util.Try
 import scala.util.control.NonFatal
 
 /**
- * `ZPure[S1, S2, R, E, A]` is a purely functional description of a computation
+ * `ZPure[S1, S2, W, R, E, A]` is a purely functional description of a computation
  * that requires an environment `R` and an initial state `S1` and may either
  * fail with an `E` or succeed with an updated state `S2` and an `A`. Because
  * of its polymorphism `ZPure` can be used to model a variety of effects
  * including context, state, and failure.
  */
```

Comment:
I would like to suggest this small change to the description :)
```suggestion
 * `ZPure[W, S1, S2, R, E, A]` is a purely functional description of a computation
 * that requires an environment `R` and an initial state `S1`, writes to a log `W` and may either
 * fail with an `E` or succeed with an updated state `S2` and an `A`. Because
 * of its polymorphism `ZPure` can be used to model a variety of effects
 * including context, state, and failure.
 */
```

Reply:
@arnoldlacko Good catch! Fixed now.